blocks/bricks/__init__.py

```diff
@@ -345,15 +345,18 @@ def __init__(self, input_dim, output_dim, num_pieces, **kwargs):
         self.output_dim = output_dim
         self.num_pieces = num_pieces
 
-        self.linear_transformation = Linear(
-            name=self.name + '_linear_to_maxout', input_dim=input_dim,
-            output_dim=output_dim * num_pieces, weights_init=self.weights_init,
-            biases_init=self.biases_init, use_bias=self.use_bias)
-        self.maxout_transformation = Maxout(name=self.name + '_maxout',
-                                            num_pieces=num_pieces)
+        self.linear_transformation = Linear(name=self.name +
```

Comment:
Just looked at your example of bad behaviour one more time and found it not so surprising. One can not expect Blocks to be so smart! It is natural that assignments to a child's configuration are not reflected in the parent's configuration. So maybe we should not change anything here altogether.